### PR TITLE
Fix  infinite loop :(

### PR DESCRIPTION
**Pull request by mkhatib**
Fixes the embarrassing #3309 

Fixed the test to catch that case.

cc/ @erwinmombay

### CODE REVIEW

File: test/functional/test-dom.js [hunk 1]
```diff
@@ -465,6 +465,7 @@ describe('DOM', () => {
       const sibling = document.createElement('div');
       expect(dom.hasNextNodeInDocumentOrder(element)).to.be.false;
       parent.appendChild(element);
+      expect(dom.hasNextNodeInDocumentOrder(element)).to.be.false;
```
Comment (cramforce) [inline test/functional/test-dom.js]: Test fails without the dom.js change?

Comment (mkhatib) [inline test/functional/test-dom.js]: Yes. Karma disconnects with an error because of the infinite loop.

Comment (jridgewell) [inline test/functional/test-dom.js]: We need a grandparent to trigger the infinite loop.

Comment (mkhatib) [inline test/functional/test-dom.js]: Basically the tree now is `parent -> child` and it goes into the loop the first time there are no siblings to the element, it changes current element to the parent and goes into the loop the parent doesn't have a next sibling and now we're in an infinite loop (without this change).

Comment (jridgewell) [inline test/functional/test-dom.js]: 👍

Comment (mkhatib) [inline test/functional/test-dom.js]: No need for grandparent. We only need one parentNode (that neither the element or the parent satisfies the nextSibling condition).

Happy to add some more if you feel this is not enough.

**Comment by cramforce** [comment]
LGTM

**Comment by dvoytenko** [comment]
Is this re-pushed into canary?

**Comment by erwinmombay** [comment]
@dvoytenko yes. i am cutting new canary release

**Pull request merged**